### PR TITLE
Enhance UCD package traceability back to the Git provider

### PR DESCRIPTION
**Pull request by dennis-behm**
This addresses #78 . It allows to automatically establish links back to the git and ci systems within the UrbanCode Deploy component version which is registered.

Based on the new `repositoryInfoPropertiesFile` configuration, which provides the prefix for different git providers, the script will extract the commit from the BuildReport.json. It relies on the zAppBuild PR https://github.com/IBM/dbb-zappbuild/pull/95, however backward compatibility is given!

A sample invocation is given in the README.

### CODE REVIEW

File: Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy [hunk 1]
```diff
@@ -124,7 +142,14 @@ def xml = new MarkupBuilder(writer)
...
 	//top level property will be added as version properties
 	//requires UCD v6.2.6 and above
+	// Url to DBB Build result
 	property(name : buildResult.getGroup() + "-buildResultUrl", value : buildResult.getUrl())
+	// Url to CI pipeline
+	if (properties.pipelineURL)property(name : "ci-pipeline-url", value : properties.pipelineURL )
+	// Populate build result properties
+	if (buildResultProperties != null) buildResultProperties.each{
+		property(name:it.key, value:it.value)
+	}
```
Comment (jbyibm) [inline Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy:152]: @dennis-behm 
Just something that come in my mind. Senthil pinged us in the past saying that when properties in this area have specific names they are duplicated from version to version. Maybe you can handle this in your PR?
image

**Review by jbyibm — CHANGES_REQUESTED** [review]

File: Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy (hunk 1 above)
Comment (dennis-behm) [reply Pipeline/CreateUCDComponentVersion/dbb-ucd-packaging.groovy]: @jbyibm yes, this is something, I can include in this as well. Capturing the gitBranch as an optional property might also be interesting.

The buildResultProperties keys should not vary, so we should be ok with publishing them in there.

Can you have another look?

**Review by jbyibm — APPROVED** [review]
Look good

**Comment by dennis-behm** [comment]
### Documentation
Applying these changes along with the updates in zAppBuild, you will be able to establish links to the dbb build result, the ci-pipeline build result and to the git commit for each individiual artifact:

### Screenshot 
image

**Pull request merged**